cfgov/unprocessed/js/apps/financial-well-being/fwb-questions.js

```diff
@@ -1,83 +1,169 @@
+  const questionElems = document.querySelectorAll( '#quiz-form fieldset' );
+  const radioButtons = document.querySelectorAll( '#quiz-form [type="radio"]' );
+  const submitButton = document.querySelector( '#submit-quiz' );
+  const questionStates = {};
+
+  /**
+   * Checks the selection status of a group of radio inputs,
+   * belonging to a parent fieldset
+   * @param {HTMLNode} questionRadios An array of dom elements.
+   * @returns {boolean} The status of the group of inputs
+   */
+  function checkQuestionState( questionRadios ) {
+    let radioIsChecked = false;
+
+    for ( let i = 0, l = questionRadios.length; i < l; i++ ) {
+      if ( questionRadios[i].checked ) {
+        radioIsChecked = true;
+      }
+    }
+
+    return radioIsChecked;
+  }
+
+  /**
+   * Checks the status of each fieldset to determine if
+   * the user has completed the form
+   * @returns {boolean} The completion status of the form
+   */
+  function checkformCompletion() {
+    let formCompleted = true;
+
+    Object.keys( questionStates ).forEach( function( key ) {
+      if ( !questionStates[key] ) {
+        formCompleted = false;
+      }
+    } );
+
+    return formCompleted;
+  }
+
+  /**
+   * Event handler to prevent clicking the submit button before
+   * the form is completed
+   * @param {MouseEvent} event The event object for the mousedown event.
+   */
+  function handleDisabledSubmit( event ) {
+    event.preventDefault();
+  }
 
-  var totalQuestions = 12;
-  var quizComplete = false;
-  var questionValues = {};
-  var radioButtons = document.querySelectorAll( '[type="radio"]' );
-  var submitButton = document.querySelector( '#submit-quiz' );
 
-  function allQuestionsCompleted() {
-    return Object.keys( questionValues ).length === totalQuestions;
+  /**
+   * Sets the disabled state of the submit button
+   */
+  function disableSubmit() {
+    submitButton.disabled = true;
+    submitButton.addEventListener( 'click', handleDisabledSubmit );
   }
 
+  /**
+   * Sets the enabled state of the submit button
+   */
   function enableSubmit() {
     submitButton.title = 'Get your score';
-
-    submitButton.classList.remove( 'a-btn__disabled' );
+    submitButton.removeAttribute( 'disabled' );
+    submitButton.removeEventListener( 'click', handleDisabledSubmit );
   }
 
+
+  /**
+   * Updates the status of the fieldset in the data store
+   * and checks if the form has been completed
+   * @param {HTMLNode} input A dom element
+   */
   function handleRadio( input ) {
     if ( input.name && input.checked ) {
-
-      questionValues[input.name] = input.value;
+      questionStates[input.name] = true;
     }
 
-    quizComplete = allQuestionsCompleted();
-
-    if ( quizComplete ) {
+    if ( checkformCompletion() ) {
       enableSubmit();
     }
   }
 
+  /**
+   * Sends the user interaction to Analytics
+   * @param {string} action A GTM data-set
+   * @param {string} label A GTM data-set
+   * @param {string} category A GTM data-set
```

Comment:
"data-set" sounds kinda weird here, since they're all just single strings

Reply:
Isn't data-set just another name for data attr?

Reply:
I guess technically dataset is all the data- attributes together https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset

Reply:
[`dataset`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) in JS is the collection of all `data-` attributes on an element, but these are just individual pieces of data.

Reply:
Ah right, thanks.